### ISSUE TITLE
🐛 BUG: @astrojs/partytown errors out, saying 'filename' must be a file URL object

### ISSUE THREAD

**Issue by okikio** [withastro/astro#4126]
### What version of `astro` are you using?

1.0.0-rc.3

### Are you using an SSR adapter? If so, which one?

None

### What package manager are you using?

pnpm

### What operating system are you using?

Linux

### Describe the Bug

`@astrojs/partytown` does this whenever I try to use it, 
```bash
> astro build

 error   The argument 'filename' must be a file URL object, file URL string, or absolute path string. Received undefined
  Code:
      36 |           })
    > 37 |         );
         |                                                 ^
      38 |       },
      39 |       "astro:build:done": async ({ dir }) => {
      40 |         await copyLibFiles(fileURLToPath(new URL("~partytown", dir)), {
    
  Stacktrace:
    at new NodeError (node:internal/errors:371:5)
    at createRequire (node:internal/modules/cjs/loader:1218:11)
    at Object.<anonymous> (/workspaces/okikio/node_modules/.pnpm/@astrojs+partytown@0.1.9/node_modules/@astrojs/partytown/dist/index.js:37:49)
    at Module._compile (node:internal/modules/cjs/loader:1101:14)
    at Object.loader (/workspaces/okikio/node_modules/.pnpm/tsm@2.2.2/node_modules/tsm/require.js:1:1187)
    at Module.load (node:internal/modules/cjs/loader:981:32)
    at Function.Module._load (node:internal/modules/cjs/loader:822:12)
    at Module.require (node:internal/modules/cjs/loader:1005:19)
    at require (node:internal/modules/cjs/helpers:102:18)
    at Object.apply (/workspaces/okikio/astro.config.ts:1:292)

 ELIFECYCLE  Command failed with exit code 1.
```

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/github-rsg1jx?file=src/layouts/Layout.astro

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by okikio** [comment]
It seems this specific error is fixed in the latest version of Astro `1.0.0-rc.4`, but there is another problem

**Comment by delucis** [comment]
Thanks @okikio! If this bug is fixed, I'll close this out and let you open a new issue or reopen with a new reproduction for the new problem 🙌
